types/providers.d.ts

```diff
@@ -29,13 +29,13 @@ export interface OAuthConfig<P extends Record<string, unknown> = Profile>
   scope: string
   params: { grant_type: string }
   accessTokenUrl: string
-  requestTokenUrl: string
+  requestTokenUrl?: string
   authorizationUrl: string
-  profileUrl: string
-  profile(profile: P, tokens: TokenSet): Awaitable<User & { id: string }>
+  profileUrl?: string
+  profile?(profile: P, tokens: TokenSet): Awaitable<User & { id: string }>
```

Comment:
While technically you are right and the docs does mention that these are optional, not providing them will cause the app to crash. See #1065 and #1846

Reply:
Yes, you're right. It's happened to me as well.

Is it a bug or do you think we should update the docs?
I'm going to revert the changes for now.

Reply:
it is a bug, - although seems to be a very rare case when you don't need the user data - and I would like to address it in the above mentioned RFC. so for now just leave it as is